src/com/lftechnology/java/training/sanish/application/model/dao/UserDao.java

```diff
@@ -0,0 +1,291 @@
+        try {
+            String query = "UPDATE users SET " + setString + " WHERE " + condition;
+            PreparedStatement preparedStatement = DbConnect.getDbConnection().prepareStatement(query);
+            try {
+                for (int i = 0; i < parameters.length; i++) {
+                    preparedStatement.setString(i + 1, parameters[i].toString());
+                }
+            } catch (SQLException e) {
+                LOGGER.log(Level.WARNING, Constants.EXCEPTION_ERROR_MSG_LABEL + "{0}", new Object[] { e });
+            }
+
+            int effectedRow = preparedStatement.executeUpdate();
+            preparedStatement.close();
+            return effectedRow;
+        } catch (SQLException e) {
+            LOGGER.log(Level.WARNING, Constants.EXCEPTION_ERROR_MSG_LABEL + "{0}", new Object[] { e });
+            DbConnect.dbClose();
+        }
+
+        return 0;
+    }
+
+    @Override public boolean setPassword(User user, String password) {
+        try {
+            DateFormat dateFormat = new SimpleDateFormat(Constants.DB_DATE_FORMAT);
+            Date date = new Date();
+            String query = "UPDATE users SET password=?, modifiedAt=? WHERE userId=?";
+            PreparedStatement preparedStatement = DbConnect.getDbConnection().prepareStatement(query);
+            preparedStatement.setString(1, password);
+            preparedStatement.setString(2, dateFormat.format(date));
+            preparedStatement.setInt(3, user.getUserId());
+            int effectedRow = preparedStatement.executeUpdate();
+            preparedStatement.close();
+            if (effectedRow != 0) {
+                return true;
+            }
+        } catch (SQLException e) {
+            LOGGER.log(Level.WARNING, Constants.EXCEPTION_ERROR_MSG_LABEL + "{0}", new Object[] { e });
+            DbConnect.dbClose();
+        }
+
+        return false;
+    }
+
+    @Override public Employee getEmployee(User user) {
+        try {
+            String query = "SELECT * FROM employees WHERE userId=?";
+            PreparedStatement preparedStatement = DbConnect.getDbConnection().prepareStatement(query);
+            preparedStatement.setInt(1, user.getUserId());
+            ResultSet rs = preparedStatement.executeQuery();
+            Employee employee = null;
+            if (rs.next()) {
+                employee = new Employee();
+                employee.setResultSetAttributes(rs);
+                ;
+            }
+
+            rs.close();
+            preparedStatement.close();
+            return employee;
+        } catch (SQLException e) {
+            LOGGER.log(Level.WARNING, Constants.EXCEPTION_ERROR_MSG_LABEL + "{0}", new Object[] { e });
+            DbConnect.dbClose();
+        }
+
+        return null;
+    }
+
+    @Override public List<UserEmployee> searchEmployee(String searchKey) {
+        List<UserEmployee> userEmployeeList = new ArrayList<UserEmployee>();
+        try {
+            searchKey += "%";
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Introduce a new variable instead of reusing the parameter "searchKey". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1226)